cf/latest-snapshot/src/index.ts

```diff
@@ -8,92 +9,139 @@ function basename(path: string) {
-			headers.set('Content-Length', `${r.end - r.start + 1}`);
-			status = 206; // Partial Content
-		} else {
-			headers.set('Content-Length', object.size.toString());
+			break;
+		}
+		default: {
+			return new Response('Invalid Snapshot Type. Only archive OR latest is supported', {
+				status: 400,
+			});
 		}
+	}
 
-		object.writeHttpMetadata(headers);
-		headers.set('etag', object.httpEtag);
-		headers.set('Accept-Ranges', 'bytes');
-		const encoded_name = encodeURIComponent(basename(object.key));
-		// Tell browsers and aria2c which filename to use. For 'wget', you have to use `--trust-server-names`.
-		headers.set(
-			'Content-Disposition',
-			`attachment; filename*=UTF-8''${encoded_name}; filename="${encoded_name}"`
-		);
+	const headers = new Headers();
 
-		if ('body' in object) {
-			return new Response(object.body, {
-				headers,
-				status,
-			});
-		} else {
-			return new Response(null, {
-				headers,
-				status,
+	// If the client requested a range, then we need to set the Content-Range header.
+	// The `parseRange` returns an error code if the range is not satisfiable (or not specified),
+	// so we can handle the range request only if it returns an array.
+	let status = 200;
+	const range = parseRange(object.size, req_headers.get('range') || '');
+	if (Array.isArray(range)) {
+		// R2Object doesn't support multiple ranges, so we only use the first one.
+		// Throw an error if there are more than one range.
+		if (range.length > 1) {
+			return new Response('Multiple ranges are not supported', {
+				status: 416, // Range Not Satisfiable
 			});
 		}
+
+		const r = range[0];
+		headers.set('Content-Range', `${r.type} ${r.start}-${r.end}/${object.size}`);
+		headers.set('Content-Length', `${r.end - r.start + 1}`);
+		status = 206; // Partial Content
+	} else {
+		headers.set('Content-Length', object.size.toString());
+	}
+
+	object.writeHttpMetadata(headers);
+	headers.set('etag', object.httpEtag);
+	headers.set('Accept-Ranges', 'bytes');
+	const encoded_name = encodeURIComponent(basename(object.key));
+	// Tell browsers and aria2c which filename to use. For 'wget', you have to use `--trust-server-names`.
+	headers.set(
+		'Content-Disposition',
+		`attachment; filename*=UTF-8''${encoded_name}; filename="${encoded_name}"`
+	);
+
+	if ('body' in object) {
+		return new Response(object.body, {
+			headers,
+			status,
+		});
+	} else {
+		return new Response(null, {
+			headers,
+			status,
+		});
 	}
 }
 
 export default {
 	async fetch(request: Request, env: Env): Promise<Response> {
-		const url = new URL(request.url);
-		const chain = (url.pathname.match(/\/latest\/(\w*)/) || ['undefined'])[1];
-
 		// Disallow any other request method except HEAD and GET, they are not sensible in the context
 		// of fetching a snapshot.
 		switch (request.method) {
-			case 'HEAD':
-			case 'GET':
-				return await get_latest(request.headers, env, chain);
-			default:
+			case 'HEAD': {
+				break;
+			}
```

Comment:
Is this the correct behavior for HEAD requests?

Reply:
Previosuly it was a fallthrough to get the same code path as GET, now we're breaking without any actions. I don't think it's correct, unless I'm missing something.